app/src/main/java/io/homeassistant/companion/android/util/vehicle/GridItems.kt

```diff
@@ -132,40 +138,57 @@ fun getDomainList(
             null
         ).getIcon(carContext)
 
-        listBuilder.addItem(
-            GridItem.Builder().apply {
-                if (icon != null) {
-                    setImage(
-                        CarIcon.Builder(
-                            IconicsDrawable(carContext, icon)
-                                .apply {
-                                    sizeDp = 64
-                                }.toAndroidIconCompat()
-                        )
-                            .setTint(CarColor.DEFAULT)
-                            .build()
-                    )
-                }
+        val entityList = allEntities.map {
+            it.values.filter { entity ->
+                entity.domain == domain && RegistriesDataHandler.getHiddenByForEntity(
+                    entity.entityId,
+                    entityRegistry
+                ) == null
             }
-                .setTitle(friendlyDomain)
-                .setOnClickListener {
-                    Log.i(TAG, "Domain:$domain clicked")
-                    screenManager.push(
-                        EntityGridVehicleScreen(
-                            carContext,
-                            serverManager,
-                            serverId,
-                            prefsRepository,
-                            serverManager.integrationRepository(serverId.value),
-                            friendlyDomain,
-                            domains,
-                            allEntities.map { it.values.filter { entity -> entity.domain == domain } },
-                            allEntities
-                        ) { }
-                    )
+        }
+        var domainIsEmpty = false
+        lifecycleScope.launch {
+            entityList.collect {
+                domainIsEmpty = it.isEmpty()
+            }
+        }

```

Comment:
Why collect here, as this doesn't invalidate the UI and cause an update?

Reply:
the entities are actually collected previously in the `init {}` here we are just building a small list to see if the domain should be hidden the last time `invalidate()` was called elsewhere.

Reply:
But this doesn't 'block' the function and it will continue with the initial `false` value / `launch` means the `domainIsEmpty` variable is probably only updated after `onGetTemplate` is already done?

Reply:
the code as is actually works. I did not have any `input_button` entities so I created one and saw the domain category, then I hid the entity and upon reloading the page the domain was hidden as well

Reply:
OK this works but I don't understand why. Even tried removing live state updates yet it still kept working.